app/com/arpnetworking/rollups/RollupGenerator.java

```diff
@@ -113,12 +116,15 @@ public RollupGenerator(
             @Named("RollupManager") final ActorRef rollupManager,
             final KairosDbClient kairosDbClient,
             final Clock clock,
-            final PeriodicMetrics metrics) {
+            final PeriodicMetrics metrics,
+            final MetricsFactory metricsFactory
+    ) {
         _metricsDiscovery = metricsDiscovery;
         _rollupManager = rollupManager;
         _kairosDbClient = kairosDbClient;
         _clock = clock;
         _metrics = metrics;
```

Comment:
Can you rename these to `_periodicMetrics` and `periodicMetrics` respectively to disambiguate from any `Metrics` created from `metricsFactory`?